### PR TITLE
"lineup" component style

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | - |
| Decisions | Improved lineup style and also added prop "nrColumns" that allows to define how many columns should the lineup component have. |
| Animated GIF | **Nr Columns:**<br>lineup_nrColumns**Responsive:**<br>lineup_responsive |

### CODE REVIEW

File: vue/components/ui/organisms/listing/listing.vue [hunk 1]
```diff
@@ -246,6 +247,9 @@ export const Listing = {
         searchWidth: {
             type: Number,
             default: 304
+        },
+        lineupNrColumns: {
+            type: Number
```
Comment (gcandal) [inline vue/components/ui/organisms/listing/listing.vue]: Either `required: true` or `default: something`. In this case let's go with `default: null`, delegating to the lineup whatever the default should be.

File: vue/components/ui/molecules/filter/filter.vue [hunk 2]
```diff
@@ -109,6 +110,9 @@ export const Filter = {
         useQuery: {
             type: Boolean,
             default: false
+        },
+        lineupNrColumns: {
+            type: Number
```
Comment (gcandal) [inline vue/components/ui/molecules/filter/filter.vue]: Either `required: true` or `default: something`. In this case let's go with `default: null`, delegating to the lineup whatever the default should be.

File: vue/components/ui/molecules/lineup/lineup.vue [hunk 3]
```diff
@@ -85,56 +86,50 @@
     text-align: left;
 }
 
-.lineup > .lineup-item {
+.lineup .lineup-item {
```
Comment (gcandal) [inline vue/components/ui/molecules/lineup/lineup.vue]: Why the change?

File: vue/components/ui/organisms/listing/listing.vue (hunk 1 above)
Comment (NFSS10) [reply vue/components/ui/organisms/listing/listing.vue]: Setting it `default: null `will make the prop in lineup be `null`. That's why I didn't added a default to the "lineupNrColumns" prop, as I wanted the default value of "nrColumns" in "lineup" component be controlled by itself.

File: vue/components/ui/molecules/filter/filter.vue (hunk 2 above)
Comment (NFSS10) [reply vue/components/ui/molecules/filter/filter.vue]: Same as said before

> Setting it default: null will make the prop in lineup be null. That's why I didn't added a default to the "lineupNrColumns" prop, as I wanted the default value of "nrColumns" in "lineup" component be controlled by itself.

**Comment by gcandal** [comment]
@NFSS10 we'll need you to make the same adaptation you did to RIPE Pulse to other projects (Sputnik, Orchestra, White) too.

File: vue/components/ui/molecules/lineup/lineup.vue (hunk 3 above)
Comment (NFSS10) [reply vue/components/ui/molecules/lineup/lineup.vue]: Redid the style, no particular reason.

Comment (gcandal) [reply vue/components/ui/molecules/lineup/lineup.vue]: Revert it then, let's keep following the principle of least action. Unless something is a clear improvement we'd be better to avoid changing it.

**Comment by gcandal** [comment]
@NFSS10 some deleted `>`s are still polluting the diff:
image
